cli/azd/pkg/azapi/standard_deployments.go

```diff
@@ -711,48 +705,16 @@ func (ds *StandardDeployments) ValidatePreflightToSubscription(
 			Tags:     tags,
 		}, nil)
 	if err != nil {
-		return validatePreflightError(rawResponse, err, "subscription")
+		return fmt.Errorf("validating deployment to subscription:\n\nValidation Error Details:\n%w", err)
 	}
 	_, err = validateResult.PollUntilDone(ctx, nil)
 	if err != nil {
-		return validatePreflightError(rawResponse, err, "subscription")
+		return fmt.Errorf("validating deployment to subscription:\n\nValidation Error Details:\n%w", err)
 	}
 
 	return nil
 }
 
-func validatePreflightError(
-	rawResponse *http.Response,
-	err error,
-	typeMessage string,
-) error {
-	title := "Validation Error Details"
-	var respErr *azcore.ResponseError
-	if errors.As(err, &respErr) {
-		err = responseToDeploymentError(title, respErr)
-	} else if err != nil {
```

Comment:
Does it make sense to keep this line of code: `err = responseToDeploymentError(title, respErr)`?

I believe this converts `azcore.ResponseError` into an `AzureDeploymentError` which is useful for azd.